### DISCUSSION TITLE
Collecting sponsorship data

### DISCUSSION THREAD

**Discussion by PraharshitaK-IS** [community/community#63831] [Programming Help]
### Select Topic Area

Question

### Body

I am trying to collect sponsorship related data on developers for an academic paper.
While the GraphQL query seems to work fine, when i run it Via POSTMAN or a Python script, i get the following reponse:

{
    "data": {
        "user": {
            "sponsorsListing": null,
            "login": "kovidgoyal"
        }
    }
}

Since I am getting the login back, I know that there is no problem with query formation or the bearer token.
Is there any reason why I shouldn't be able to access sponsor listing related data outside of the GraphQL explorer?

**Comment by oftharsh** [comment]
GitHub's GraphQL API allows you to access various types of data, including sponsorship-related information. However, there are certain considerations and limitations that might affect your ability to access certain data outside of the GraphQL Explorer:

1. **Scope of Access**: Some data in GitHub's API might have restricted access based on your authentication token's scope or the user's privacy settings. It's possible that sponsorship-related data falls under such restrictions.

2. **Permissions**: Make sure that the authenticated user (the one associated with the provided bearer token) has the necessary permissions to access sponsorship-related data. Users might need to have certain roles or permissions enabled to view sponsorship information.

3. **Rate Limits**: GitHub's API has rate limits that restrict the number of requests you can make in a given time period. If you're using Postman or a Python script, make sure you're not hitting the rate limits, which could prevent you from retrieving data.

4. **Data Availability**: Not all users might have sponsorship listings, so some users might return `null` for the `sponsorsListing` field even if they have other data available.

5. **Data Consistency**: Data availability and consistency can sometimes vary between different parts of the GitHub platform, including the API. It's possible that the GraphQL Explorer might have access to certain data that is not yet available or fully exposed through other methods.

6. **API Version**: Ensure that you are using the correct version of the GraphQL API. GitHub occasionally introduces changes or updates to the API, which could affect the availability of certain data.

7. **Caching**: GitHub's API employs caching mechanisms to improve performance. It's possible that the GraphQL Explorer's data is being fetched from a different cache compared to your Postman or Python script.

To troubleshoot further, you can try the following steps:

- Double-check your authentication token and ensure it has the necessary scopes and permissions.
- Review the user's privacy settings and verify that the sponsorship information is set to be publicly accessible.
- Experiment with different users to see if you can retrieve sponsorship data for them.
- Monitor the API response headers for rate limit information.

If you continue to face issues, you might consider reaching out to GitHub support or their community forums for more specific assistance related to accessing sponsorship data via the GraphQL API.

**Comment by PraharshitaK-IS** [comment]
Hi Harsha,
From my understanding of the documentation Sponsors data does not require any permissions - https://docs.github.com/en/rest/overview/permissions-required-for-fine-grained-personal-access-tokens?apiVersion=2022-11-28

I have tried it for a lot of developers - who have confirmed sponsor listings (scraped from the github site)
API versioning also doesn't seem to be the issue as I am hitting the latest.

**Comment by PraharshitaK-IS** [comment]
Could you please help me find this "GitHub support or their community forums ".
I'm not sure where I can get assistance with this.
